infrastructure/singlesubject/test_singlesubject.py

```diff
@@ -1,53 +1,17 @@
-import sys
-# import pytest
+import sys, pytest, singlesubject
 from pathlib import Path
-import singlesubject
 
-# it is assumed to run within the infrastructure/singlesubject folder
 test_path = Path('tests')
+test_dirs = [d for d in test_path.iterdir() if d.is_dir() and d.name.startswith('test')]
 
-def test1(tmp_path):
-    singlesubject.main(['exe', test_path / 'test1', tmp_path / 'test1/singlesubject1', '1'])
-    singlesubject.main(['exe', test_path / 'test1', tmp_path / 'test1/singlesubject2', '2'])
-    singlesubject.main(['exe', test_path / 'test1', tmp_path / 'test1/singlesubject3', '3'])
+@pytest.mark.parametrize('test_dir', test_dirs)
+def test_singlesubject(test_dir, tmp_path):
+    subject_numbers = ['1', '2', '3']
+    # "Usage: singlesubject.py <inputdir> <outputdir> <participant_nr>"
+    for participant_nr in subject_numbers:
+        print()
+        singlesubject.main(['exe', str(test_dir), str(tmp_path / f'{test_dir.name}/singlesubject-output-{participant_nr}'), participant_nr])
 
-    # FIXME determine the correct counts and then enable these tests
```

Comment:
removing this block of comments does not mean that there should not be more strict tests. Please look at the outputs and implement the appropriate counts on the numbers of files of specific types.